compiler/rustc_middle/src/ty/query/plumbing.rs

```diff
@@ -166,10 +166,10 @@ impl<'tcx> TyCtxt<'tcx> {
             }
         });
 
-        if num_frames != None {
-            eprintln!("we're just showing a limited slice of the query stack");
-        } else {
+        if num_frames == None || num_frames >= Some(i) {
```

Comment:
The output did not change... Not sure what's wrong. If `num_frames` is `Some(2)` and `i` is `0`, this should return true (according to the playground at least: https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=57ab1b6049e331f483455f1ce170e183)